src/Form/formatters.js

```diff
@@ -163,14 +163,25 @@ const getMMAndYYYYFromString = (expiration) => {
  * @param {InternalDataStorageObject} credentials
  * @return {boolean}
  */
-const shouldStoreCredentials = ({ credentials }) => Boolean(credentials.password);
+const shouldStoreCredentials = ({ credentials }, hasOnlyOneCredential) => {
+    if (credentials.password) {
+        return Boolean(credentials.password);
+    } else {
+        return hasOnlyOneCredential && Boolean(credentials.username);
+    }
```

Comment:
This seems overly complex. In fact, is this even needed anymore? If credentials are either password or username or both, regardless of `hasOnlyOneCredential` this should always be true. It's either:

- A lone password → save
- A lone username → save
- Both username and password → save

Am I missing anything?

Reply:
I think I was just trying to be extra protective and cover only the cases we need (which is specifically one credential), I don't think there's any weird case. We can just remove it, good idea.